great_expectations/experimental/datasources/postgres_datasource.py

```diff
@@ -43,7 +44,7 @@ def param_defaults(self, data_asset: DataAsset) -> Dict[str, List]:
 
 @pydantic_dc.dataclass(frozen=True)
 class SqlYearMonthSplitter(ColumnSplitter):
-    method_name: str = "split_on_year_and_month"
+    method_name: Literal["split_on_year_and_month"] = "split_on_year_and_month"
```

Comment:
This is the substantive change (combined with the updated `column_splitter` type union) that allows loading the specific splitter subclass from config.